easybuild/framework/easyconfig/tweak.py

```diff
@@ -836,6 +837,12 @@ def map_easyconfig_to_target_tc_hierarchy(ec_spec, toolchain_mapping, targetdir=
     # Determine the name of the modified easyconfig and dump it to target_dir
     ec_filename = '%s-%s.eb' % (parsed_ec['ec']['name'], det_full_ec_version(parsed_ec['ec']))
     tweaked_spec = os.path.join(targetdir or tempfile.gettempdir(), ec_filename)
+    if os.path.exists(tweaked_spec):
+        if build_option('force'):
+            print_warning("Overwriting existing file at %s with tweaked easyconfig file (due to --force)", tweaked_spec)
+        else:
+            raise EasyBuildError("A file already exists at %s where tweaked easyconfig file would be written",
+                                 tweaked_spec)
```

Comment:
@mboisson Shouldn't we do this in `dump` itself instead? We may be using `dump` is other places in the future...

Reply:
In this case I really think doing it in `dump` itself makes more sense...

I'm not sure why there are two code paths where one uses `write_file` and another uses `dump`, maybe @ocaisa knows?

Reply:
There are two paths because one uses regex substitution and the other modifies the EasyConfig object and dumps the modified object.

Ultimately the dump method does use `write_file` so perhaps the question is whether we want to pass down options, the choices currently are:
```
def write_file(path, txt, append=False, forced=False, backup=False):
    """
    Write given contents to file at given path;
    overwrites current file contents without backup by default!

    :param path: location of file
    :param txt: contents to write to file
    :param append: append to existing file rather than overwrite
    :param forced: force actually writing file in (extended) dry run mode
    :param backup: back up existing file before overwriting or modifying it
    """
```

Reply:
Perhaps doing a backup is sufficient? That way nothing ever gets removed and no need for the force option

Reply:
Would just need to add `backup=True` to https://github.com/easybuilders/easybuild-framework/blob/develop/easybuild/framework/easyconfig/easyconfig.py#L800 to switch this on by default (and can't think of a good reason you wouldn't want to)

Reply:
I think write_file should at the very least issue a warning if not fail if it is going to be over-writing an existing file.

Reply:
If a backup is created, it will print a warning message for that.

But I agree with @mboisson here, it shouldn't overwrite existing files without `--force` (and it should still create a backup if `--force` is used).

Reply:
@mboisson W.r.t. making `write_file` fail hard rather than overwriting an existing file: we can't do that without breaking backward compatibility, changing that would likely break a lot of easyblocks. Also, always having to use force to make `write_file` actually overwrite existing file would be a bit much, that's not standard semantics in Linux either...

Reply:
So, how do we proceed? Are you saying that the original check that @mboisson had is appropriate then and maybe couple that with adding the backup kw in the dump?

Reply:
I took a look at it, the best way is to enhance `write_file` a bit to optionally not blindly overwrite existing files, and then use `write_file` correctly in both places.

Done in https://github.com/ComputeCanada/easybuild-framework/pull/8